spec.bs

```diff
@@ -3054,40 +3057,67 @@ A <dfn>server auction response</dfn> is a [=struct=] with the following [=struct
   : <dfn>is chaff</dfn>
   :: A [=boolean=] field indicating whether this response should be ignored.
   : <dfn>ad render url</dfn>
-  :: The [=URL=] of the ad for the winning bid.
+  :: [=URL=]. The [=leading bid info/leading bid=]'s [=generated bid/ad descriptor=]'s [=ad descriptor/url=] from the
+     auction executed on the trusted auction server.
   : <dfn>ad components</dfn>
-  :: A [=list=] of [=URLs=] containing the ad components for the winning bid.
+  :: A [=list=] of [=URLs=]. A list of [=leading bid info/leading bid=]'s
+     [=generated bid/ad component descriptors=]'s [=ad descriptor/urls=] from the
+     auction executed on the trusted auction server.
   : <dfn>interest group name</dfn>
-  :: The winning interest group's name as a [=string=].
+  :: The [=leading bid info/leading bid=]'s [=generated bid/interest group=]'s
+     [=interest group/name=] as a [=string=] from the
+     auction executed on the trusted auction server.
   : <dfn>interest group owner</dfn>
-  :: The winning interest group's owner [=origin=].
+  :: An [=origin=]. The [=leading bid info/leading bid=]'s
+     [=generated bid/interest group=]'s [=interest group/owner=] from the
+     auction executed on the trusted auction server.
   : <dfn>bidding groups</dfn>
-  :: A [=list=] of interest group owner and interest group name pairs that bid
-     in this auction.
+  :: A [=list=] of interest group [=interest group/owner=] and
+     interest group [=interest group/name=] pairs that bid
+     in the auction executed on the trusted auction server.
   : <dfn>score</dfn>
-  :: {{double}} or Null. TODO
+  :: Null or {{double}}. Null when the server auction is not a component auction.
+     For component auctions, contains the [=leading bid info/top score=]
+     from the component auction executed on the trusted auction server.
   : <dfn>bid</dfn>

```

Comment:
I guess we can combine the bid and bid currency fields to a single field of [=bid with currency=]?

Reply:
Done. That differs from the implementation, but should be cleaner and is functionally equivalent.

Reply:
this is the internal struct (like our C++ code) that we can define in whatever way we want, as long as we create it correctly from parsing response. It'll be good to make it match the impl, but not necessary when there is a better way to spec it. 
The data type that the spec must match the impl is web/user facing data, like IDL, response, etc,.